course/models/models.py

```diff
@@ -378,6 +378,28 @@ def get_decoded_results(self):
         stdout = base64.b64decode(self.results[0]['stdout'] or "").decode('utf-8')
         return parse_junit_xml(stdout)
 
+    def get_formatted_test_results(self):
+        return str(len(self.get_passed_test_results())) + "/" + str(self.get_num_tests())
+
+    def get_passed_test_results(self):
+        all_tests = self.get_decoded_results()
+        return list(filter(get_passed_tests, all_tests))
+
+    def get_failed_test_results(self):
+        all_tests = self.get_decoded_results()
+        return list(filter(get_failed_tests, all_tests))
+
+    def get_num_tests(self):
+        return len(self.get_decoded_results())
+
+
+def get_failed_tests(test):
```

Comment:
The function should be here in models.py. I would recommend using the lambda function to implicitly define these in the filter.